src/gui/appclient.cpp

```diff
@@ -645,6 +628,90 @@ void AppClient::askUserToLoginAgain(int userDbId, QString userEmail, bool invali
     _gui->openLoginDialog(userDbId, invalidTokenError);
 }
 
+bool AppClient::startServer(bool restartClient) {
+    // Start the client
+    QString pathToExecutable = QCoreApplication::applicationDirPath();
+
+#if defined(Q_OS_WIN)
+    pathToExecutable += QString("/%1.exe").arg(APPLICATION_EXECUTABLE);
+#else
+    pathToExecutable += QString("/%1").arg(APPLICATION_EXECUTABLE);
+#endif
+
+    QStringList arguments;
+    if (!restartClient) {
+        arguments << QString("--commPort") << QString::number(_commPort) << QString("--noStartClient");
+    }
+
+    QProcess *clientProcess = new QProcess(this);
+    clientProcess->setProgram(pathToExecutable);
+    clientProcess->setArguments(arguments);
+    
+    clientProcess->startDetached();
+
+    if (restartClient) {
+        QTimer::singleShot(0, qApp, SLOT(quit()));
+        return true;
+    } 
+    return true;
+}
+
+bool AppClient::connectToServer() {
+
+    // Check if a commPort is provided
+    if (_commPort == 0) {
+		QMessageBox msgBox;
+		msgBox.setText(tr("No comm port provided! Clean restart in progress..."));
+		msgBox.exec();
+        displayHelpText(tr("No comm port provided! Clean restart in progress..."));
+        startServer(true /*Close the client and let the server restart it*/);
+		return false;
+	}
+
+    // Connect to server
+    int count = 0;
+    while (!CommClient::instance()->connectToServer(_commPort) && count < CONNECTION_TRIALS) {
+        std::cout << "Connection to server failed!" << std::endl;
+        count++;
+    }
+
+    if (count == CONNECTION_TRIALS) {
+        QMessageBox msgBox;
+        msgBox.setText(tr("Unable to connect to application server!"));
+        msgBox.exec();
+        //startServer(true /*Close the client and let the server restart it*/);
```

Comment:
I don't think so, for me this is a totally different issue.
Herve's fix aims to automatically restart the app in case on crash.
This part of code handle the case where the server takes too much time to reply. Not ideal perhaps but unrelated to this PR. What do you think?